src/Files.App/Helpers/QuickLookHelpers.cs

```diff
@@ -1,35 +1,68 @@
-using Files.Shared.Extensions;
-using System.Diagnostics;
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Security.Principal;
 using System.Threading.Tasks;
-using Windows.Foundation.Collections;
 
-namespace Files.App.Helpers
+namespace Files.App.Helpers;
+
+public static class QuickLookHelpers
 {
-    public static class QuickLookHelpers
+    public static Task ToggleQuickLook(IShellPage associatedInstance, bool switchPreview = false)
     {
-        public static async Task ToggleQuickLook(IShellPage associatedInstance, bool switchPreview = false)
+        if (!App.AppModel.IsQuickLookSupported || !associatedInstance.SlimContentPage.IsItemSelected || associatedInstance.SlimContentPage.IsRenamingItem)
+            return null;
```

Comment:
Should be `return Task.CompletedTask;` I guess, or it will crash if you await it.